Libraries/LibGUI/Application.h

```diff
@@ -53,8 +54,9 @@ class Application : public Core::Object {
     void register_global_shortcut_action(Badge<Action>, Action&);
     void unregister_global_shortcut_action(Badge<Action>, Action&);
 
-    void show_tooltip(const StringView&, const Gfx::IntPoint& screen_location);
+    void show_tooltip(const StringView&, const Gfx::IntPoint& screen_location, const RefPtr<Widget> tooltip_source_widget);
     void hide_tooltip();
+    RefPtr<Widget> tooltip_source_widget() const { return m_tooltip_source_widget; };
```

Comment:
No need to return a `RefPtr` here let the caller decide if he wants to participate in ownership of the widget.
```suggestion
    Widget* tooltip_source_widget() { return m_tooltip_source_widget; }
```